### DISCUSSION TITLE
Leveraging custom variable set in _quaro.yml in a filter script?

### DISCUSSION THREAD

**Discussion by dweng0** [quarto-dev/quarto-cli#6429] [Q&A] [answered]
### Description

Hi All, 

I have created an [extension](https://github.com/clearmatics/qreacto) that allows you to inject react components into md/qmd documents. The script that does this also checks for things like `.css` and supporting `.js`/`.ts` files.

Where the script "looks" for css/ts/js files is currently hardcoded [here](https://github.com/clearmatics/qreacto/blob/main/_extensions/qreacto/qreacto.lua#L3-L5)
``` lua
local component_folder = '_components'
local resources_folder = '_components'
```
 I would like to be able to use a custom _quarto metadata variable and leverage that instead

``` lua 
local component_folder =  quarto.???.component-location
local resources_folder = quarto.???.resource-location
```
I haven't seen any information about this in the documentation but was wondering if it is something that can be achieved?

**Comment by mcanouil** [comment, marked as answer]
All metadata are in `Meta` as described in Pandoc's documentation (which is referenced in Quarto's documentation).
See:
- https://quarto.org/docs/extensions/filters#development

See the following extension which uses and adds meta data: <https://github.com/mcanouil/quarto-lua-env>.
